compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangLiteral.java

```diff
@@ -88,7 +88,7 @@ public NodeKind getKind() {
     @Override
     public String toString() {
         // To distinguish between `()` and `null`
-        return value == null ? originalValue : String.valueOf(value);
+        return value == null ? (originalValue == null ? "null" : originalValue) : String.valueOf(value);
```

Comment:
Is this the correct fix? Isn't the original check intended to cover exactly this scenario? Do we know why `value` is not null but `originalValue` is null?

I believe this is related to https://github.com/ballerina-platform/ballerina-lang/issues/34588.

Reply:
Both `originalValue` and `Value` gets `null` when reading, since ` BIRTypeWriter` does not write when the type tag is `NIL`

Reply:
`null` or `"null"`?

Reply:
`null`

Reply:
My original comment should have been

> Do we know why value is `null` but `originalValue` is not `"null"`?

Ideally, the behaviour should be the same in normal compilation and via BIR symbol enter.

Reply:
Actually I think we have an issue differentiating between `"null"` and `null` anyway, which is probably also the reason for https://github.com/ballerina-platform/ballerina-lang/issues/34588.

With this fix, wouldn't we see this issue further?

Reply:
Current implementation neglect writing value when the type is `NIL`
https://github.com/ballerina-platform/ballerina-lang/blob/fdb8d28e9454125877cb5405043dd92c7faf754b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L562

Reply:
Yeah IMO differentiating "null" and null is different issue compared to this

> Actually I think we have an issue differentiating between `"null"` and `null` anyway, which is probably also the reason for #34588.
> 
> With this fix, wouldn't we see this issue further?

Reply:
> Current implementation neglect writing value when the type is `NIL`
> 
> https://github.com/ballerina-platform/ballerina-lang/blob/fdb8d28e9454125877cb5405043dd92c7faf754b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L562

Even then we can create the value properly (as what was there when generating the BIR) at BIR symbol enter, right?